pkg/kubelet/cm/container_manager_linux.go

```diff
@@ -1060,7 +1060,29 @@ func isKernelPid(pid int) bool {
 	return err != nil && os.IsNotExist(err)
 }
 
+// GetCapacity returns node capacity data for "cpu", "memory", "ephemeral-storage", and "huge-pages*"
+// At present this method is only invoked when introspecting ephemeral storage
 func (cm *containerManagerImpl) GetCapacity() v1.ResourceList {
+	// We store allocatable ephemeral-storage in the capacity property once we Start() the container manager
+	if _, ok := cm.capacity[v1.ResourceEphemeralStorage]; !ok {
```

Comment:
Need we check the feature gate status here like below?
https://github.com/kubernetes/kubernetes/blob/5de1a754c8f875cacfa49b9c5ed63c6c35b7e9e5/pkg/kubelet/cm/container_manager_linux.go#L640-L649

Reply:
The feature gate check is in place before this method is invoked:

https://github.com/kubernetes/kubernetes/blob/v1.21.0/pkg/kubelet/nodestatus/setters.go#L332

Should we add the feature gate check inside the actual method itself as well?

Reply:
Yes, because a refactor might accidentally have something else call this.

https://github.com/kubernetes/kubernetes/blob/588d98440706e317ca0fcc5b510285caaf161e51/pkg/kubelet/cm/container_manager_linux.go#L640-L648

also has the feature flag.

Reply:
Makes sense, done. (UT updated as well)